common/ast/src/parser/statement.rs

```diff
@@ -858,6 +884,8 @@ pub fn statement(i: Input) -> IResult<StatementMsg> {
         rule!(
             #create_share: "`CREATE SHARE [IF NOT EXISTS] <share_name> [ COMMENT = '<string_literal>' ]`"
             | #drop_share: "`DROP SHARE [IF EXISTS] <share_name>`"
+            | #grant_share_object: "`GRANT ShareGrantObjectPrivilege ON ShareGrantObject TO SHARE <share_name>`"
```

Comment:
Please replace `ShareGrantObjectPrivilege` with easy to understand content like `{ USAGE | SELECT | REFERENCE_USAGE }`

So does for `ShareGrantObject`